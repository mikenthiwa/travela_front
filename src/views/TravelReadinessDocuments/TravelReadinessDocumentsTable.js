import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { isEmpty } from 'lodash';
import { Link } from 'react-router-dom';
import withLoading from '../../components/Hoc/withLoading';
import './TravelReadinessDocuments.scss';

export class UserReadiness extends Component {
  render(){
    const { name, userId, department, passportsCount, visasCount, othersCount } = this.props;
    return (
      <tr className="table__rows">
        <td className="mdl-data-table__cell--non-numeric table__data readiness__cell-name freeze table__data-pointer">
          <Link className="table__data--link button-outline table__id-pointer" to={`/travel-readiness/${userId}`}>{name}</Link>
        </td>
        <td className="mdl-data-table__cell--non-numeric table__data">{department}</td>
        <td className="mdl-data-table__cell--non-numeric table__data">{passportsCount}</td>
        <td className="mdl-data-table__cell--non-numeric table__data">{visasCount}</td>
        <td className="mdl-data-table__cell--non-numeric table__data">{othersCount}</td>
      </tr>
    );
  }
}

export const TableHead = () => (
  <thead>
    <tr>
      <th className="mdl-data-table__cell--non-numeric table__head freeze request_id">Name</th>
      <th className="mdl-data-table__cell--non-numeric table__head pl-sm-100d description-left">Department</th>
      <th className="mdl-data-table__cell--non-numeric table__head pl-sm-100d description-left">No. of Passports</th>
      <th className="mdl-data-table__cell--non-numeric table__head">No. of Visas</th>
      <th className="mdl-data-table__cell--non-numeric table__head">No. of Other Documents</th>
    </tr>
  </thead>
);

export const TableBody = ({ users }) => (
  <tbody className="table__body  approvals_table_body">
    {
      users.map(user => {
        const visasCount = user.travelDocuments.filter(document => document.type === 'visa').length;
        const passportsCount = user.travelDocuments.filter(document => document.type === 'passport').length;
        const othersCount = user.travelDocuments.filter(document => document.type === 'other').length;
        return (
          <UserReadiness
            key={user.userId}
            name={user.fullName}
            userId={user.userId}
            department={user.department}
            visasCount={visasCount}
            passportsCount={passportsCount}
            othersCount={othersCount}
          />
        );
      })
    }
  </tbody>
);

const ReadinessTable = ({ users }) => (
  <div className="table__container">
    { isEmpty(users)
      ? <h1 id="no-results">No results</h1>
      : (
        <table className="mdl-data-table mdl-js-data-table readiness-table">
          <TableHead />
          <TableBody users={users} />
        </table>
      )
    }
  </div>
);

ReadinessTable.propTypes = {
  users: PropTypes.array.isRequired,
};

UserReadiness.propTypes = {
  name: PropTypes.string.isRequired,
  userId: PropTypes.string.isRequired,
  department: PropTypes.string,
  passportsCount: PropTypes.number.isRequired,
  visasCount: PropTypes.number.isRequired,
  othersCount: PropTypes.number.isRequired
};

UserReadiness.defaultProps = {
  department: ''
};

TableBody.propTypes = {
  users: PropTypes.array.isRequired,
};

export default withLoading(ReadinessTable);
