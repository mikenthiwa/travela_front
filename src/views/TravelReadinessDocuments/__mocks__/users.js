const users = [
  {
    'id': 1200,
    'fullName': 'John Snow',
    'userId': '-LJlsimdclse9s',
    'department': 'Success',
    'createdAt': '2018-08-16T11:11:52.181Z',
    'updatedAt': '2018-08-16T11:11:52.181Z',
    'travelDocuments': [{
      'id': 'b9gnYkdzG',
      'type': 'visa',
      'data': {
        'country': 'Kenya',
        'entryType': 'H-2A',
        'expiryDate': '06-01-2018',
        'dateOfIssue': '02-01-2018',
        'cloudinaryUrl': 'https://res.cloudinary.com/ined/image/upload/v1538568663/Logo_blue_2x.png'
      },
      'isVerified': false,
      'createdAt': '2019-01-04T11:11:52.181Z',
      'updatedAt': '2019-01-16T11:11:52.181Z',
    },
    {
      'id': 'pk42DLnx4C',
      'type': 'passport',
      'data': {
        'expiryDate': '2029-11-01',
        'dateOfBirth': '1970-01-01',
        'dateOfIssue': '2018-11-01',
        'nationality': 'kenyan',
        'placeOfIssue': 'Kenya',
        'cloudinaryUrl': 'https://res.cloudinary.com/dbk8ky2',
        'passportNumber': 'qw357etrty'
      },
      'isVerified': false,
      'createdAt': '2019-01-04T11:11:52.181Z',
      'updatedAt': '2019-01-16T11:11:52.181Z',
    },
    {
      'id': 'SyOyr_AtC',
      'type': 'visa',
      'data': {
        'country': 'Kenya',
        'entryType': 'H-2A',
        'expiryDate': '06-01-2018',
        'dateOfIssue': '02-01-2018',
        'cloudinaryUrl': 'https://res.cloudinary.com/ined/image/upload/v1538568663/Logo_blue_2x.png'
      },
      'isVerified': true,
      'createdAt': '2019-01-04T11:11:52.181Z',
      'updatedAt': '2019-01-08T06:38:35.139Z',
    }
    ]
  },
  {
    'id': 1,
    'fullName': 'Celestine Ekoh-Ordan',
    'userId': '-LMIzC-bCc10w7Uqc7-A',
    'department': 'Talent & Development',
    'createdAt': '2019-01-08T06:37:34.532Z',
    'updatedAt': '2019-01-08T08:14:37.897Z',
    'travelDocuments': []
  },
  {
    'id': 2,
    'fullName': 'Celestine1 Ekoh-Ordan',
    'userId': '-LMIzC-bcfffffCc10w7Uqc7-A',
    'department': 'Talent & Development',
    'createdAt': '2019-01-08T06:37:34.532Z',
    'updatedAt': '2019-01-08T08:14:37.897Z',
    'travelDocuments': []
  },
  {
    'id': 3,
    'fullName': 'Celestine2 Ekoh-Ordan',
    'userId': '-LMIzC-bcgsdvcc10w7Uqc7-A',
    'department': 'Talent & Development',
    'createdAt': '2019-01-08T06:37:34.532Z',
    'updatedAt': '2019-01-08T08:14:37.897Z',
    'travelDocuments': []
  },
  {
    'id': 4,
    'fullName': 'Black Window',
    'userId': '-LMIzC-bCc1csah0w7Uqc7-A',
    'department': 'Talent & Development',
    'createdAt': '2019-01-08T06:37:34.532Z',
    'updatedAt': '2019-01-08T08:14:37.897Z',
    'travelDocuments': []
  },
  {
    'id': 5,
    'fullName': 'Black Window',
    'userId': '-LMIzC-bCcsch10w7Uqc7-A',
    'department': 'Talent & Development',
    'createdAt': '2019-01-08T06:37:34.532Z',
    'updatedAt': '2019-01-08T08:14:37.897Z',
    'travelDocuments': []
  },
  {
    'id': 6,
    'fullName': 'Black Window',
    'userId': '-LMIzC-bcvgCc10w7Uqc7-A',
    'department': 'Talent & Development',
    'createdAt': '2019-01-08T06:37:34.532Z',
    'updatedAt': '2019-01-08T08:14:37.897Z',
    'travelDocuments': []
  },
  {
    'id': 7,
    'fullName': 'Black Window',
    'userId': '-LMIzC-bCc10cxzw7Uqc7-A',
    'department': 'Talent & Development',
    'createdAt': '2019-01-08T06:37:34.532Z',
    'updatedAt': '2019-01-08T08:14:37.897Z',
    'travelDocuments': []
  },
  {
    'id': 8,
    'fullName': 'Black Window',
    'userId': '-LMIzC-bCccsvzhx10w7Uqc7-A',
    'department': 'Talent & Development',
    'createdAt': '2019-01-08T06:37:34.532Z',
    'updatedAt': '2019-01-08T08:14:37.897Z',
    'travelDocuments': []
  },
  {
    'id': 9,
    'fullName': 'Black Window',
    'userId': '-LMIzC-bCcscdh10w7Uqc7-A',
    'department': 'Talent & Development',
    'createdAt': '2019-01-08T06:37:34.532Z',
    'updatedAt': '2019-01-08T08:14:37.897Z',
    'travelDocuments': []
  },
  {
    'id': 10,
    'fullName': 'Black Window',
    'userId': '-LMIzC-bCc10wvd7Uqc7-A',
    'department': 'Talent & Development',
    'createdAt': '2019-01-08T06:37:34.532Z',
    'updatedAt': '2019-01-08T08:14:37.897Z',
    'travelDocuments': []
  },
  {
    'id': 11,
    'fullName': 'Black Window',
    'userId': '-LMIzC-bdshvCc10w7Uqc7-A',
    'department': 'Talent & Development',
    'createdAt': '2019-01-08T06:37:34.532Z',
    'updatedAt': '2019-01-08T08:14:37.897Z',
    'travelDocuments': []
  },
  {
    'id': 12,
    'fullName': 'Black Window',
    'userId': '-LMIzC-bCc1sdhvc0w7Uqc7-A',
    'department': 'Talent & Development',
    'createdAt': '2019-01-08T06:37:34.532Z',
    'updatedAt': '2019-01-08T08:14:37.897Z',
    'travelDocuments': []
  }
];

export default users;
