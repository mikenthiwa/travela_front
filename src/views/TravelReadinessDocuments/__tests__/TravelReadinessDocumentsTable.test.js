import React from 'react';
import { shallow } from 'enzyme';
import { TableHead, TableBody, UserReadiness } from '../TravelReadinessDocumentsTable';
import users from '../__mocks__/users';

describe('TravelReadinessDocumentsTable', () => {
  it('renders TableHead', () => {
    const wrapper = shallow(<TableHead />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders TableBody', () => {
    const wrapper = shallow(<TableBody users={users} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders UserReadiness', () => {
    const props = {
      name: 'John Doe',
      department: 'TDD',
      passportsCount: 4,
      visasCount: 2,
      othersCount:3,
      userId: 'xlkdsf'
    };
    const wrapper = shallow(<UserReadiness {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
