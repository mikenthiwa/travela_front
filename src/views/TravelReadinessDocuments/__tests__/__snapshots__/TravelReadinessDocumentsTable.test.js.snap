// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TravelReadinessDocumentsTable renders TableBody 1`] = `
<tbody
  className="table__body  approvals_table_body"
>
  <UserReadiness
    department="Success"
    key="-LJlsimdclse9s"
    name="John Snow"
    othersCount={0}
    passportsCount={1}
    userId="-LJlsimdclse9s"
    visasCount={2}
  />
  <UserReadiness
    department="Talent & Development"
    key="-LMIzC-bCc10w7Uqc7-A"
    name="Celestine Ekoh-Ordan"
    othersCount={0}
    passportsCount={0}
    userId="-LMIzC-bCc10w7Uqc7-A"
    visasCount={0}
  />
  <UserReadiness
    department="Talent & Development"
    key="-LMIzC-bcfffffCc10w7Uqc7-A"
    name="Celestine1 Ekoh-Ordan"
    othersCount={0}
    passportsCount={0}
    userId="-LMIzC-bcfffffCc10w7Uqc7-A"
    visasCount={0}
  />
  <UserReadiness
    department="Talent & Development"
    key="-LMIzC-bcgsdvcc10w7Uqc7-A"
    name="Celestine2 Ekoh-Ordan"
    othersCount={0}
    passportsCount={0}
    userId="-LMIzC-bcgsdvcc10w7Uqc7-A"
    visasCount={0}
  />
  <UserReadiness
    department="Talent & Development"
    key="-LMIzC-bCc1csah0w7Uqc7-A"
    name="Black Window"
    othersCount={0}
    passportsCount={0}
    userId="-LMIzC-bCc1csah0w7Uqc7-A"
    visasCount={0}
  />
  <UserReadiness
    department="Talent & Development"
    key="-LMIzC-bCcsch10w7Uqc7-A"
    name="Black Window"
    othersCount={0}
    passportsCount={0}
    userId="-LMIzC-bCcsch10w7Uqc7-A"
    visasCount={0}
  />
  <UserReadiness
    department="Talent & Development"
    key="-LMIzC-bcvgCc10w7Uqc7-A"
    name="Black Window"
    othersCount={0}
    passportsCount={0}
    userId="-LMIzC-bcvgCc10w7Uqc7-A"
    visasCount={0}
  />
  <UserReadiness
    department="Talent & Development"
    key="-LMIzC-bCc10cxzw7Uqc7-A"
    name="Black Window"
    othersCount={0}
    passportsCount={0}
    userId="-LMIzC-bCc10cxzw7Uqc7-A"
    visasCount={0}
  />
  <UserReadiness
    department="Talent & Development"
    key="-LMIzC-bCccsvzhx10w7Uqc7-A"
    name="Black Window"
    othersCount={0}
    passportsCount={0}
    userId="-LMIzC-bCccsvzhx10w7Uqc7-A"
    visasCount={0}
  />
  <UserReadiness
    department="Talent & Development"
    key="-LMIzC-bCcscdh10w7Uqc7-A"
    name="Black Window"
    othersCount={0}
    passportsCount={0}
    userId="-LMIzC-bCcscdh10w7Uqc7-A"
    visasCount={0}
  />
  <UserReadiness
    department="Talent & Development"
    key="-LMIzC-bCc10wvd7Uqc7-A"
    name="Black Window"
    othersCount={0}
    passportsCount={0}
    userId="-LMIzC-bCc10wvd7Uqc7-A"
    visasCount={0}
  />
  <UserReadiness
    department="Talent & Development"
    key="-LMIzC-bdshvCc10w7Uqc7-A"
    name="Black Window"
    othersCount={0}
    passportsCount={0}
    userId="-LMIzC-bdshvCc10w7Uqc7-A"
    visasCount={0}
  />
  <UserReadiness
    department="Talent & Development"
    key="-LMIzC-bCc1sdhvc0w7Uqc7-A"
    name="Black Window"
    othersCount={0}
    passportsCount={0}
    userId="-LMIzC-bCc1sdhvc0w7Uqc7-A"
    visasCount={0}
  />
</tbody>
`;

exports[`TravelReadinessDocumentsTable renders TableHead 1`] = `
<thead>
  <tr>
    <th
      className="mdl-data-table__cell--non-numeric table__head freeze request_id"
    >
      Name
    </th>
    <th
      className="mdl-data-table__cell--non-numeric table__head pl-sm-100d description-left"
    >
      Department
    </th>
    <th
      className="mdl-data-table__cell--non-numeric table__head pl-sm-100d description-left"
    >
      No. of Passports
    </th>
    <th
      className="mdl-data-table__cell--non-numeric table__head"
    >
      No. of Visas
    </th>
    <th
      className="mdl-data-table__cell--non-numeric table__head"
    >
      No. of Other Documents
    </th>
  </tr>
</thead>
`;

exports[`TravelReadinessDocumentsTable renders UserReadiness 1`] = `
<tr
  className="table__rows"
>
  <td
    className="mdl-data-table__cell--non-numeric table__data readiness__cell-name freeze table__data-pointer"
  >
    <Link
      className="table__data--link button-outline table__id-pointer"
      replace={false}
      to="/travel-readiness/xlkdsf"
    >
      John Doe
    </Link>
  </td>
  <td
    className="mdl-data-table__cell--non-numeric table__data"
  >
    TDD
  </td>
  <td
    className="mdl-data-table__cell--non-numeric table__data"
  >
    4
  </td>
  <td
    className="mdl-data-table__cell--non-numeric table__data"
  >
    2
  </td>
  <td
    className="mdl-data-table__cell--non-numeric table__data"
  >
    3
  </td>
</tr>
`;
